internal/rcm/api_test.go

```diff
@@ -0,0 +1,129 @@
+package rcm_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"net"
+	"net/http"
+	"regexp"
+	"testing"
+
+	"github.com/google/uuid"
+	test_distro "github.com/osbuild/osbuild-composer/internal/distro/test"
+	"github.com/osbuild/osbuild-composer/internal/rcm"
+	"github.com/osbuild/osbuild-composer/internal/store"
+)
+
+func TestBasicRcmAPI(t *testing.T) {
+	// Test the HTTP API responses
+	// This test mainly focuses on HTTP status codes and JSON structures, not necessarily on their content
+	socket := "localhost:10999"
+
+	var cases = []struct {
+		Method            string
+		Path              string
+		Body              string
+		ContentType       string
+		ExpectedStatus    int
+		ExpectedBodyRegex string
+	}{
+		{"GET", "/submit", ``, "", http.StatusMethodNotAllowed, ``},
+		{"POST", "/submit", `{"status":"RUNNING"}`, "application/json", http.StatusBadRequest, ``},
+		{"POST", "/submit", `{"status":"RUNNING"}`, "text/plain", http.StatusBadRequest, ``},
+		{"POST", "/submit", `{"distribution": "fedora-30", "architectures":["x86_64"], "repositories": ["aaa"]}`, "application/json", http.StatusOK, `{"compose_id":".*"}`},
+		{"POST", "/status", `{"status":"RUNNING"}`, "application/json", http.StatusNotFound, ``},
+		{"POST", "/status/111-222-333", `{"status":"RUNNING"}`, "application/json", http.StatusMethodNotAllowed, ``},
+		{"GET", "/status/7802c476-9cd1-41b7-ba81-43c1906bce73", `{"status":"RUNNING"}`, "application/json", http.StatusBadRequest, `{"error_reason":"Compose UUID does not exist"}`},
+	}
+
+	ln, err := net.Listen("tcp", socket)
```

Comment:
I would prefer these tests not to rely on TCP - I think their main purpose is to test the routes, not the transport layer. We can reuse the code from weldr API tests.

It totally makes sense to test also if the TCP listening works as expected but I think it's more suitable to test it in integration tests.

Reply:
This seemed like the easiest way to implement this, but I can check the helpers you wrote.

Reply:
Even without the helpers it might be easier. :)